From ef83390129cef5daa22b9c5ceb28f0c06bbfb5d4 Mon Sep 17 00:00:00 2001
From: Clement Calmels <clement.calmels@intel.com>
Date: Thu, 21 Mar 2013 10:06:32 +0100
Subject: [PORT FROM MAIN] Trace invalid property length

BZ: 94604

This patch adds an error trace when trying to set a property with bad
name or value length.

Change-Id: I6df2d347102d25b9c60462fe3a80518e8438cff6
Orig-Change-Id: Ib5821ed4ce299a3e8a14bd575139cd28c3018699
Signed-off-by: Clement Calmels <clement.calmels@intel.com>
Reviewed-on: http://android.intel.com:8080/97749
Reviewed-by: cactus <cactus@intel.com>
Reviewed-by: Du, Chenyang <chenyang.du@intel.com>
Reviewed-by: Chen, HuaqiangX <huaqiangx.chen@intel.com>
Reviewed-by: Yao, Yong <yong.yao@intel.com>
Tested-by: Chen, HuaqiangX <huaqiangx.chen@intel.com>
Reviewed-by: buildbot <buildbot@intel.com>
Tested-by: buildbot <buildbot@intel.com>
---
 init/property_service.c |    7 ++++---
 1 files changed, 4 insertions(+), 3 deletions(-)

diff --git a/init/property_service.c b/init/property_service.c
index e3c9733..5fb3ff0 100755
--- a/init/property_service.c
+++ b/init/property_service.c
@@ -345,9 +345,10 @@ int property_set(const char *name, const char *value)
     size_t namelen = strlen(name);
     size_t valuelen = strlen(value);
 
-    if(namelen >= PROP_NAME_MAX) return -1;
-    if(valuelen >= PROP_VALUE_MAX) return -1;
-    if(namelen < 1) return -1;
+    if(namelen >= PROP_NAME_MAX || valuelen >= PROP_VALUE_MAX || namelen < 1) {
+        ERROR("sys_prop: invalid property %s[%d] %s[%d]\n", name, namelen, value, valuelen);
+        return -1;
+    }
 
     pi = (prop_info*) __system_property_find(name);
 
-- 
1.7.4.1

